DxDispatch/src/dxdispatch/main.cpp

```diff
@@ -46,6 +34,25 @@ int main(int argc, char** argv)
     auto d3dModule = std::make_shared<D3d12Module>();
     auto dmlModule = std::make_shared<DmlModule>();
 
+    if (args.PrintHelp())
+    {
+        LogInfo(args.HelpText());
+        return 0;
+    }
+
+    if (args.ShowDependencies())
+    {
+#if defined(_WIN32) && !defined(_GAMING_XBOX)
+        // D3D12.dll lazily loads D3D12Core.dll. Calling any exported function forces D3D12Core.dll to load
+        // so its version can be printed, and GetDebugInterface is inexpensive.
+        Microsoft::WRL::ComPtr<ID3D12Debug> debug;
+        d3dModule->GetDebugInterface(IID_PPV_ARGS(&debug));
+#endif
+
+        PrintDependencies();
+        return 0;
```

Comment:
>return 0; [](http://example.com/codeflow?start=8&length=9)

Since this returns early, if you also pass a JSON file to execute, you won't get an error - it will just (appears to) silently not execute it. Similarly if you pass `show_dependencies` and `show_adapters`, the show adapters will be silently ignored.

How about continuining onward to line 86, except just don't throw an error?

```
if has ModelPath
{
    // do model stuff
}
else if not (ShowDependencies or ShowAdapters)
{
}
```